ta/sims/ta_sims.c

```diff
@@ -136,3 +136,41 @@ TEE_Result sims_get_counter(void *session_context, uint32_t param_types,
 
 	return TEE_SUCCESS;
 }
+
+TEE_Result sims_entry_panic(void *session_context, uint32_t param_types,
+			    TEE_Param params[4])
+{
+	uint32_t ret_orig;
+	TEE_Result res = TEE_SUCCESS;
+	struct sims_session *ctx = (struct sims_session *)session_context;
+
+	if (param_types != TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
+					   TEE_PARAM_TYPE_NONE,
+					   TEE_PARAM_TYPE_NONE,
+					   TEE_PARAM_TYPE_NONE))
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	if (params[0].memref.buffer && params[0].memref.size) {
+		res = sims_open_ta_session(session_context,
+					   param_types, params);
+		if (res != TEE_SUCCESS)
+			return res;
+
+		params[0].memref.buffer = NULL;
+		params[0].memref.size = 0;
+
+		/* Trigger panic to remote TA */
+		TEE_InvokeTACommand(ctx->sess, 0,
+				    TA_SIMS_CMD_PANIC,
+				    param_types, params,
+				    &ret_orig);
+	} else {
+		TEE_Panic(0xbeef); /* Trigger panic to current TA */
+	}
+
+	/*
+	 * Should not be reached, but if it is the testsuite can detect that
+	 * TEE_Panic() returned instead of panicking the TA.
+	 */
```

Comment:
I think you can remove this comment. If the TA has invoked another TA, the former TA will return without error.